### DISCUSSION TITLE
UART packet size

### DISCUSSION THREAD

**Discussion by sheershak-agarwal** [nasa/fprime#1023] [Q&A] [answered]
Hello,

I am using UART to talk to my GPS but the uart message by gps is quite long. So, F-prime uart drivers cut them in smaller chunks like this:
Packet Received(16): #TIMEA,COM1,0,4.
Packet Received(16): 0,UNKNOWN,0,4.00
Packet Received(16): 0,024c0020,9924,
Packet Received(16): 16410;INVALID,0.
Packet Received(16): 000000000,0.0003
Packet Received(16): 33564,0.00000000
Packet Received(16): 000,0,0,0,0,0,0

I took a look at the Linux Driver Implementation and changed the cfg.c_cc[VMIN] = 0;  cfg.c_cc[VTIME] = 10; parameters to make the packets longer. But it doesn't how big the VMIN value is, I can not received the whole message in one packet.

What are your thoughts on this? 

p.s. I am already working on a message builder program that will take chunks and build a main message.

**Comment by timcanham** [comment, marked as answer]
That's how we've done it in the past. Never assume you are getting the full packet because of driver buffering, and write your logic to assemble the full packet across receive calls.

**Reply by LeStarch** [reply]
I will back this up.  It is hard to guarantee you'll ever get a full packet as UART is asynchronous....so there is always a chance that your read aligns poorly with the remote side.  It is best to reconstruct packets across multiple receives.
